source/Mlos.SettingsSystem.CodeGen/CodeWriters/CSharpCodeWriter.cs

```diff
@@ -24,11 +24,16 @@ public override void WriteBeginFile()
             // Tell stylecop to ignore this file.
             //
             WriteLine("// <auto-generated />");
+
+            // Tell stylecop to *really* ignore some documentation requirement rules.
+            //
+            WriteLine("#pragma warning disable CS1591");

```

Comment:
Assuming that the external project may wish to turn on *all* of the stylecop rules, then we need to make sure our code will compile with them.
For whatever reason, `// <auto-generated />` doesn't handle this particular rule (which is about providing documentation for all fields), so we have to disable it as a shortcut for now.
Obviously in the future it would be nice to have even the generated code have reasonable documentation comment headers on it.